src/textual/app.py

```diff
@@ -562,7 +562,20 @@ def __init__(
         """Number of lines to scroll in the Y direction with wheel or trackpad."""
 
         self._installed_screens: dict[str, Screen | Callable[[], Screen]] = {}
+        for var_name, dct in (("SCREENS", self.SCREENS), ("MODES", self.MODES)):
+            for screen_name, v in dct.items():
+                if (
+                    isinstance(v, Screen)
+                    or (not callable(v))
+                    and (not isinstance(v, str))
+                ):
+                    raise TypeError(
+                        f"{var_name} should contain a Screen type or callable, not an instance"
+                        f" (got instance of {type(v).__name__} for {screen_name!r})"
+                    )
         self._installed_screens.update(**self.SCREENS)
+        self.modes: dict[str, str | Callable[[], Screen]] = self.MODES.copy()
```

Comment:
I think `modes` should be private. It probably isn't of much use for the dev, so `self._modes` is probably better.

If we want it to be public later, we could add a property.

Reply:
Ah, there was a test that accessed `MODES` to check changes made from `add_mode`, so I figured it was documented behavior. Will move it `_modes`.